src/targets/esp8266.ts

```diff
@@ -179,83 +185,97 @@ export default class ESP8266ROM {
+    const ets_xtal = (loader.transport.baudrate * uart_div) / 1000000 / this.XTAL_CLK_DIVIDER;
+    let norm_xtal;
+    if (ets_xtal > 33) {
+      norm_xtal = 40;
+    } else {
+      norm_xtal = 26;
+    }
+    if (Math.abs(norm_xtal - ets_xtal) > 1) {
+      loader.log(
+        "WARNING: Detected crystal freq " +
+          ets_xtal +
+          "MHz is quite different to normalized freq " +
+          norm_xtal +
+          "MHz. Unsupported crystal in use?",
+      );
+    }
+    return norm_xtal;
+  };
 
-  static _d2h(d) {
-      var h = (+d).toString(16);
-      return h.length === 1 ? '0' + h: h;
+  public _d2h(d: number) {
+    const h = (+d).toString(16);
+    return h.length === 1 ? "0" + h : h;
   }
 
-  static read_mac = async (loader) => {
-      var mac0 = await this.read_efuse(loader, 0);
-      mac0 = mac0 >>> 0;
-      var mac1 = await this.read_efuse(loader, 1);
-      mac1 = mac1 >>> 0;
-      var mac3 = await this.read_efuse(loader, 3);
-      mac3 = mac3 >>> 0;
-      var mac = new Uint8Array(6);
+  public read_mac = async (loader: ESPLoader) => {
+    let mac0 = await this.read_efuse(loader, 0);
+    mac0 = mac0 >>> 0;
+    let mac1 = await this.read_efuse(loader, 1);
+    mac1 = mac1 >>> 0;
+    let mac3 = await this.read_efuse(loader, 3);
+    mac3 = mac3 >>> 0;
+    const mac = new Uint8Array(6);
 
-      if (mac3 != 0) {
-          mac[0] = (mac3 >> 16) & 0xff;
-          mac[1] = (mac3 >> 8) & 0xff;
-          mac[2] = mac3 & 0xff;
-      }
-      else if (((mac1 >> 16) & 0xff) == 0) {
-          mac[0] = 0x18;
-          mac[1] = 0xfe;
-          mac[2] = 0x34;
-      }
-      else if (((mac1 >> 16) & 0xff) == 1) {
-          mac[0] = 0xac;
-          mac[1] = 0xd0;
-          mac[2] = 0x74;
-      }
-      else {
-          loader.log("Unknown OUI");
-      }
+    if (mac3 != 0) {
+      mac[0] = (mac3 >> 16) & 0xff;
+      mac[1] = (mac3 >> 8) & 0xff;
+      mac[2] = mac3 & 0xff;
+    } else if (((mac1 >> 16) & 0xff) == 0) {
+      mac[0] = 0x18;
+      mac[1] = 0xfe;
+      mac[2] = 0x34;
+    } else if (((mac1 >> 16) & 0xff) == 1) {
+      mac[0] = 0xac;
+      mac[1] = 0xd0;
+      mac[2] = 0x74;
+    } else {
+      loader.log("Unknown OUI");
+    }
 
-      mac[3] = (mac1 >> 8) & 0xff;
-      mac[4] = mac1 & 0xff;
-      mac[5] = (mac0 >> 24) & 0xff;
+    mac[3] = (mac1 >> 8) & 0xff;
+    mac[4] = mac1 & 0xff;
+    mac[5] = (mac0 >> 24) & 0xff;
 
-      return(this._d2h(mac[0])+":"+this._d2h(mac[1])+":"+this._d2h(mac[2])+":"+this._d2h(mac[3])+":"+this._d2h(mac[4])+":"+this._d2h(mac[5]));
-  }
+    return (
+      this._d2h(mac[0]) +
+      ":" +
+      this._d2h(mac[1]) +
+      ":" +
+      this._d2h(mac[2]) +
+      ":" +
+      this._d2h(mac[3]) +
+      ":" +
+      this._d2h(mac[4]) +
+      ":" +
+      this._d2h(mac[5])
+    );
+  };
 
-  static get_erase_size = function(offset, size) {
-      return size;
-  }
+  public get_erase_size = function (offset: number, size: number) {
```

Comment:
If you are going to change all these functions, let's set them correctly for classes:

```suggestion
  public get_erase_size(offset: number, size: number) {
```